client/client.go

```diff
@@ -3306,6 +3279,18 @@ func (c *Client) GetTaskEventHandler(allocID, taskName string) drivermanager.Eve
 	return nil
 }
 
+func (c *Client) SetAllocRunnerFactory(factory interfaces.AllocRunnerFactory) {
+	c.allocLock.Lock()
+	defer c.allocLock.Unlock()
+	c.allocrunnerFactory = factory
+}
+
+func (c *Client) newAllocRunner(conf *allocrunner.Config) (interfaces.AllocRunner, error) {
+	c.allocLock.RLock()
+	defer c.allocLock.RUnlock()
+	return c.allocrunnerFactory(conf)
+}
+
```

Comment:
Since the alloc runner is set by the client's configuration, why do we need to mutate it at runtime? Seems only useful for test cases where you know that know allocs will be run before you can swap factories.

If updating the factory at runtime is necessary, let's slap a big "For testing purposes only!" warning on it.

If updating the factory at runtime is _not_ necessary, let's kill the Set method and remove the locking from the newAR helper.